src/seedu/addressbook/data/person/Address.java

```diff
@@ -23,9 +32,15 @@
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
         if (!isValidAddress(address)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+            
         }
-        this.value = address;
+        String[] splitAddress = address.split(",");
+        block = new Block(splitAddress[0]);
+        street = new Street(splitAddress[1]);
+        unit = new Unit(splitAddress[2]);
+        postalcode = new Postalcode(splitAddress[3]);
+        
+        this.value = block.getblockName() + "," + street.getstreetName() + "," + unit.getunitName() + "," + postalcode.getpostalcodeNum();
```

Comment:
- Keeping this variable (value) seems redundant and may cause future problems like inconsistent data as you have introduced new ways of handling the address value.